src/Clapi/Types.hs

```diff
@@ -35,6 +45,75 @@ instance MonadFail (Either String) where
 type Attributee = String
 type Site = String
 
+class UMsg a where
+   uMsgPath :: a -> Path
+
+data UMsgError = UMsgError {errMsgPath :: Path, errMsgTxt :: T.Text} deriving (Eq, Show)
+
+instance UMsg UMsgError where
+    uMsgPath = errMsgPath
+
+data SubMessage =
+    UMsgSubscribe {subMsgPath :: Path}
+  | UMsgUnsubscribe {subMsgPath :: Path}
+  deriving (Eq, Show)
+
+instance UMsg SubMessage where
+    uMsgPath = subMsgPath
+
+-- Separate because not valid in RequestBundle
+data TreeUpdateMessage =
+    UMsgAssignType {tuMsgPath :: Path, tuMsgTypePath :: Path}
+  | UMsgDelete {tuMsgPath :: Path}
+  deriving (Eq, Show)
+
+instance UMsg TreeUpdateMessage where
+    uMsgPath = tuMsgPath
+
+data DataUpdateMessage =
+    UMsgAdd {
```

Comment:
Not that this is a problem, as I think at some point we're just going to have to go through and apply a pretty-printer to the code anyway, but the Haskelly way of laying this out is:
```haskell
    UMsgAdd
      { duMsgPath :: Path
      , duMsgTime :: Time
      ...
      duMsgsite :: (MaybeSite)
      }
    | UMsgSet
      { duMsgPath :: Path
      , du...
```
Not worth doing right now :)

Reply:
I think I was copying the style of message but can fix.